src/components/GlobalHeader/RightContent.tsx

```diff
@@ -0,0 +1,188 @@
+import React from 'react'
+import { FormattedMessage, formatMessage } from 'umi/locale'
+import { Spin, Tag, Menu, Icon, Avatar, Tooltip } from 'antd'
+import moment from 'moment'
+import groupBy from 'lodash/groupBy'
+import { NoticeIcon } from 'ant-design-pro'
+import HeaderSearch from '../HeaderSearch'
+import HeaderDropdown from '../HeaderDropdown'
+import SelectLang from '../SelectLang'
+import styles from './index.less'
+
+export default function GlobalHeaderRight (props) {
+  const {
+    currentUser,
+    fetchingNotices,
+    onNoticeVisibleChange,
+    onMenuClick,
+    onNoticeClear,
+    theme,
+    notices = [],
+    dispatch
+  } = props
+
+  function getNoticeData () {
+    if (notices.length === 0) {
+      return {}
+    }
+    const newNotices = notices.map(notice => {
+      const newNotice = { ...notice }
+      if (newNotice.datetime) {
+        newNotice.datetime = moment(notice.datetime).fromNow()
+      }
+      if (newNotice.id) {
+        newNotice.key = newNotice.id
+      }
+      if (newNotice.extra && newNotice.status) {
+        const color = {
+          todo: '',
+          processing: 'blue',
+          urgent: 'red',
+          doing: 'gold'
+        }[newNotice.status]
+        newNotice.extra = (
+          <Tag color={color} style={{ marginRight: 0 }}>
+            {newNotice.extra}
+          </Tag>
+        )
+      }
+      return newNotice
+    })
+    return groupBy(newNotices, 'type')
+  }
```

Comment:
Please move this function to the module scope. It will be easier to test. I suggest new name: `groupNotices`. 
Also please add tests. 

```
export function  groupNotices (notices) { 
   ...
}
``` 

- [ ] move getNoticeData
- [ ] add tests to getNoticeData

Reply:
BTW - I think it will be also simpler if we will use for loop:

```
let groups =  new Map<string, Array<string>>();
for (let n of notices) {
  const newNotice
  ....
  const g = groups.get(newNotice.type) || []
  g.push(newNotice)
  groups.set(newNotice.type, g)
}
```

Reply:
pushed test case and foreach to for conversion to another issue

Reply:
OK!